icebergs_framework.F90

```diff
@@ -756,6 +767,269 @@ subroutine move_berg_between_cells(bergs)  !Move icebergs onto the correct lists
 
 end subroutine move_berg_between_cells
 
+
+! #############################################################################
+
+subroutine update_halo(bergs)
+! Arguments
+type(icebergs), pointer :: bergs
+! Local variables
+type(iceberg), pointer :: kick_the_bucket, this
+integer :: nbergs_to_send_e, nbergs_to_send_w
+integer :: nbergs_to_send_n, nbergs_to_send_s
+integer :: nbergs_rcvd_from_e, nbergs_rcvd_from_w
+integer :: nbergs_rcvd_from_n, nbergs_rcvd_from_s
+type(icebergs_gridded), pointer :: grd
+integer :: i, nbergs_start, nbergs_end
+integer :: stderrunit
+integer :: grdi, grdj
+integer :: halo_width
+integer :: temp1, temp2
+  
+halo_width=bergs%grd%iceberg_halo  ! Must be less than current halo value used for updating weight.
+
+ ! Get the stderr unit number
+   stderrunit = stderr()
+
+ ! For convenience
+   grd=>bergs%grd
+
+
+
+! Step 1: Clear the current halos
+
+
+  do grdj = grd%jsd,grd%jsc-1 ;  do grdi = grd%isd,grd%ied
+    call delete_all_bergs_in_list(bergs, grdj, grdi)
+  enddo ; enddo
+
+  do grdj = grd%jec+1,grd%jed ;  do grdi = grd%isd,grd%ied
+    call delete_all_bergs_in_list(bergs,grdj,grdi)
+  enddo ; enddo
+
+  do grdj = grd%jsd,grd%jed ;    do grdi = grd%isd,grd%isc-1
```

Comment:
Double counting corners but it probably doesn't matter because it's just clearing the lists twice.